src/auth/auth.service.ts

```diff
@@ -0,0 +1,132 @@
+import { userService } from "@/user/user.service";
+import { jwtTokenService } from "@/services/JwtTokenService";
+import {
+  UnauthenticatedError,
+  UnprocessableEntityError,
+} from "@/errors/http";
+import {
+  LoginForm,
+  loginFormSchema,
+  RegistrationForm,
+  registrationFormSchema,
+} from "./auth.forms";
+import { validateExistingUser, validateUserPassword } from "./auth.validators";
+import bcrypt from "bcryptjs";
+import { prisma } from "@/db/prisma";
+
+export class AuthService {
+  async register(data: RegistrationForm) {
+    const result = registrationFormSchema.safeParse(data);
+
+    if (!result.success) {
+      throw new UnprocessableEntityError({
+        errors: result.error.flatten().fieldErrors,
+      });
+    }
+
+    const { email, password, name } = data;
+
+    await validateExistingUser(email);
+
+    const hashedPassword = await bcrypt.hash(password, 10);
+    const newUser = await userService.createUser(name, email, hashedPassword);
+    const tokenPair = jwtTokenService.generateTokenPair({ userId: newUser.id });
+
+    await prisma.refreshToken.create({
```

Comment:
This method call can throw a list of `PrismaClientKnownRequestError`s.

In this case, mostly there could be two of them with [error codes](https://www.prisma.io/docs/orm/reference/error-reference#prisma-client-query-engine): `P2002` and `P2003`.

Consider catching them (e.g. [here](https://github.com/Code-the-Dream-School/ii-practicum-team-2-back/blob/main/src/goal/goal.service.ts#L88)).